File: src/main/java/niklasu/speedtester/influx/InfluxModule.java
package niklasu.speedtester.influx;

import java.net.UnknownHostException;
import java.util.Objects;

import com.influxdb.client.InfluxDBClient;
import com.influxdb.client.InfluxDBClientFactory;
import com.influxdb.client.WriteApiBlocking;
import niklasu.speedtester.config.ConfigProvider;

public class InfluxModule {

    public static InfluxWriter getInfluxWriter(ConfigProvider configProvider) {
        String influxUrl = configProvider.influxHost();
        if (Objects.equals(influxUrl, "")) {
            return speed -> {
                // noop
            };
        }
        InfluxDBClient influxDBClient = InfluxDBClientFactory.create(influxUrl,
                configProvider.influxToken().toCharArray(), configProvider.influxOrg(), configProvider.influxBucket());
        WriteApiBlocking writeApi = influxDBClient.getWriteApiBlocking();

        return new InfluxConnector(writeApi, configProvider);
    }

    public interface InfluxWriter {
        void write(double speed) throws UnknownHostException;
    }

}

File: src/main/java/niklasu/speedtester/influx/InfluxModule.kt
package niklasu.speedtester.influx

import com.google.inject.AbstractModule
import com.google.inject.Provides
import niklasu.speedtester.config.ConfigProvider
import org.influxdb.InfluxDB
import org.influxdb.InfluxDBFactory

class InfluxModule : AbstractModule() {

    @Provides
    private fun getInfluxConnector(configProvider: ConfigProvider): InfluxWriter {
        val influxHost = configProvider.influxHost
        if (influxHost == "") {
            return object : InfluxWriter {
                override fun write(speed: Double) {
                    //noop
                }
            }
        }
        val influxDB = InfluxDBFactory.connect(influxHost)
        influxDB.setDatabase("speedtester")
        return InfluxConnector(influxDB, configProvider)
    }

    interface InfluxWriter {
        fun write(speed: Double)
    }

}